skrub/_dataframe/_common.py

```diff
@@ -640,6 +640,8 @@ def is_numeric(col):
 
 @is_numeric.specialize("pandas", argument_type="Column")
 def _is_numeric_pandas(col):
+    if pd.api.types.is_object_dtype(col):
```

Comment:
why is this change necessary? in code that calls that function to find non-numeric columns for example to try converting them to numbers this could cause these columns to be missed? how does it relate to normalization?

Reply:
maybe that extra conversion could be done in total_std_norm to make sure we don't affect other code

Reply:
I had a test failing because of this, and it was unrelated to `total_std_norm`, still not sure why. We have a similar logic for booleans in `_is_bool_pandas`, why wouldn't we want a similar behavior for numeric?

Reply:
ok maybe we want it I just want to investigate a bit more because it might affect other parts of skrub. (also it will typically cause a copy of the whole column, making what was a simple dtype check potentially expensive). for bool I guess the reason was booleans with None in an object dtype is how bools with missing values get represented